cirq-ft/cirq_ft/algos/mean_estimation/mean_estimation_operator_test.py

```diff
@@ -0,0 +1,236 @@
+# Copyright 2023 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Sequence, Tuple
+
+import cirq
+import cirq_ft
+import numpy as np
+import pytest
+from attr import frozen
+from cirq._compat import cached_property
+from cirq_ft.algos.mean_estimation import CodeForRandomVariable, MeanEstimationOperator
+from cirq_ft.infra import bit_tools
+
+
+@frozen
+class BernoulliSynthesizer(cirq_ft.PrepareOracle):
```

Comment:
Are these worth putting in a class rather than in a test?

Reply:
Umm, I think they are fine in tests right now. The goal of the `MeanEstimationOperator` is to construct the mean estimation unitary given a use-case specific `synthesizer` and `encoder` and I am not aware of any practical use cases for a bernoulli encoder / synthesizer except for testing that the mean estimation code is working as intended. Also, the classes would need to be rewritten in a much nicer way for them to move out of the tests into main source code.